doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/Xhtml5BaseSink.java

```diff
@@ -1658,6 +1659,35 @@ public void tableHeaderCell( SinkEventAttributes attributes )
     private void tableCell( boolean headerRow, MutableAttributeSet attributes )
     {
         Tag t = ( headerRow ? HtmlMarkup.TH : HtmlMarkup.TD );
+        
+        if (   !headerRow 
+            && null != this.cellCountStack
```

Comment:
this.foo != null
is far more common and unsurprising. Putting the literal first is a C convention not needed in Java.

Reply:
Starting with constants in logical expressions prevents from a very common mistakes that can be very tricky, this concerns Java, C, C++, C# att least.

See [https://cwe.mitre.org/data/definitions/481.html](url)

Reply:
In C, yes, Not in Java because == is used for comparison instead of =. The only way this can occur in Java is if you're comparing to a boolean and in that case you should just use the boolean directly.